packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -186,28 +165,50 @@ export default class GraphQLRunner {
     const document = this.parse(query)
     const errors = this.validate(schema, document)
 
-    const result =
-      errors.length > 0
-        ? { errors }
-        : execute({
-            schema,
-            document,
-            rootValue: context,
-            contextValue: withResolverContext({
+    let tracer
+    if (parentSpan) {
+      tracer = new GraphQLSpanTracer(`GraphQL Query ${queryName}`, {
+        parentSpan,
+      })
+
+      tracer.start()
+    }
+
+    try {
+      const promise =
```

Comment:
This may be a promise or may not be a promise. If there were validation errors - it will be just an object. `execute` can also return `promise` OR `value`. It probably doesn't affect anything but still a bit misleading when reading.